projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts

```diff
@@ -38,25 +37,29 @@ export class AnnotationSetComponent {
   }
 
   public onAnnotationUpdate(annotation: Annotation) {
-    const annotations = this.annotationSet.annotations.filter(a => a.id !== annotation.id);
-
-    annotations.push(annotation);
+    this.api
+      .postAnnotation(annotation)
+      .subscribe(newAnnotation => {
+        const index = this.annotationSet.annotations.findIndex(a => a.id === newAnnotation.id);
 
-    this.annotationSet.annotations = annotations;
-    this.api.postAnnotationSet(this.annotationSet);
+        this.annotationSet.annotations[index] = newAnnotation;
+      });
   }
 
   public onAnnotationDelete(annotation: Annotation) {
-    this.annotationSet.annotations = this.annotationSet.annotations.filter(a => a.id !== annotation.id);
-    this.api.postAnnotationSet(this.annotationSet);
+    this.api
+      .deleteAnnotation(annotation.id)
+      .subscribe(() => {
+        this.annotationSet.annotations = this.annotationSet.annotations.filter(a => a.id !== annotation.id);
```

Comment:
same comment as above in terms updating the state in the service post the api call..

Reply:
potential orphan subscription. might want an ngOnDestroy to tidy these up